wgpu-rs/src/lib.rs

```diff
@@ -486,19 +595,19 @@ impl Queue {
     }
 }
 
-impl SwapChain {
-    //TODO: borrow instead of new object?
-    pub fn get_next_texture(&self) -> (Texture, TextureView) {
-        let output = wgn::wgpu_swap_chain_get_next_texture(self.id);
-        (
-            Texture {
-                id: output.texture_id,
-            },
-            TextureView { id: output.view_id },
-        )
+impl<'a> Drop for SwapChainOutput<'a> {
+    fn drop(&mut self) {
+        wgn::wgpu_swap_chain_present(*self.swap_chain_id);
```

Comment:
The swapchain output is now holding the swapchain from modification: they can't get new frames anyway. There is nothing the user can do to screw this up. They can only stop using the output and drop it.
We could totally have explicit "present", but I didn't find it useful. Open for re-consideration ;)

Reply:
I think it's a bit unexpected not to have the usual `present` or `swap_buffers` at the end of rendering a frame (although WebGL does this implicitly too). Also `drop` causes some work to be performed here, which doesn't seem very common.

This could be confusing, especially if the user is trying to map calls between the WebGPU specification and wgpu-rs and doesn't notice where the missing `present` went. Principle of least astonishment and all that :)

Reply:
That's how the W3C group agreed to do things on IDL side, interestingly.